app/controllers/index.rb

```diff
@@ -0,0 +1,63 @@
+get '/session-viewer' do
+  session.inspect
+end
+
+get '/session-reset' do
+  session[:user_password] = nil
+  session[:user_id] = nil
+  session.inspect
+end
+
+get '/' do
+  @shoes = ["La Sportiva Solution", "Mad Rock Shark", "Evolv Shaman", "Tenaya Oasi"]
+    erb :index
+end
+
+
+
+get '/logout' do
+  session.clear
+  redirect '/'
+end
+
+get '/login/?' do
+  erb :login
+end
+
+get '/login/again' do
+  erb :login_again
+end
+
+post '/login' do
+  user = User.find_by(email: params[:email])
+  if user.nil? # If email doesn't exist, try again.
+    redirect '/login/again'
+  else # If email exists, try to authenticate
+    if user.authenticate(params[:password])
+      session[:user_id] = user.id
+      session[:user_password] = user.password
```

Comment:
Never store a plain text password anywhere. Also never store a password in the session. Especially given you public display of the the session as I commented on above.